src/docs/guides/spring-boot.md

```diff
@@ -0,0 +1,186 @@
+
+To create a new Spring Boot app, ensure that you have [JDK](https://www.oracle.com/java/technologies/downloads/) installed on your machine.
+
+Go to [start.spring.io](https://start.spring.io) to initialize a new Spring Boot app. Select the options below to customize and generate your starter app.
+
+- Project: Maven
+- Language: Java
+- Spring Boot: 3.3.4
+- Project Metadata:
+    - Group: com.railwayguide
+    - Artifact: helloworld
+    - Name: helloworld
+    - Description: Demo project for Railway Guide
+    - Package name: com.railwayguide.helloworld
+    - Packaging: jar
+    - Java: 17
+- Dependencies: 
+    - Click the **Add Dependencies** button and search for **Spring Web**. Select it.
+
+![Spring Boot App Initializer](https://res.cloudinary.com/railway/image/upload/v1729619101/springboot_app_on_railway.png)
+_Config to initialize our new app_
+
+Now, click on the **Generate** button, download the zipped file and unpack it into a folder on your machine. 
+
+### Modify the Application File
+
+Next, open the app in your IDE and navigate to the `src/main/java/com/railwayguide/helloworld/HelloWorldApplication.java` file.
+
+Replace the content with the code below:
+
+```java
+package com.railwayguide.helloworld;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@SpringBootApplication
+@RestController
+public class HelloworldApplication {
+
+	public static void main(String[] args) {
+		SpringApplication.run(HelloworldApplication.class, args);
+	}
+
+	@GetMapping("/")
+    public String hello() {
+      return String.format("Hello world from Java Spring Boot!");
+    }
+
+}
+```
+
+We added a `hello()` method that returns the response: `Hello world from Java Spring Boot!`.
+
+The `@RestController` annotation designates this class as a web controller, while `@GetMapping("/")` maps the `hello()` method to handle requests sent to the root URL, `/`.
+
+### Run the Spring Boot App locally
+
+Next, `cd` into the `helloworld` directory via the terminal and run the following Maven command:
+
+```bash
+./mvnw spring-boot:run
+```
+
+**Note:** This is a Maven wrapper for Linux and macOS, which uses a bundled version of Maven from **.mvn/wrapper/maven-wrapper.jar** instead of relying on the system-installed version.
+
+Open your browser and go to `http://localhost:8080` to see your app.
+
+## Deploy the Spring Boot App to Railway
+
+Railway offers multiple ways to deploy your Spring Boot app, depending on your setup and preference. 
+
+### One-Click Deploy from a Template
+
+If you’re looking for the fastest way to get started, the one-click deploy option is ideal. 
+
+Click the button below to begin:
+
+[![Deploy on Railway](https://railway.app/button.svg)](https://railway.app/new/template/JvYvDw)
```

Comment:
I personally dont think we should link to templates from creators we don't know or don't have any contact with.

maybe you could follow your guide and publish a spring boot template under your account?

Reply:
Oh, this guy is not active like Vin? I assumed he was.

Reply:
i dont know how active they are on the platform itself, or if they even check on their templates, but they are not active in the community

Reply:
Switched with the new template I published.